src/Pos/Wallet/Web/Tracking.hs

```diff
@@ -417,8 +418,16 @@ applyModifierToWallet wid newTip CAccModifier{..} = do
     mapM_ (WS.addCustomAddress ChangeAddr . fst) (MM.insertions camChange)
     WS.getWalletUtxo >>= WS.setWalletUtxo . MM.modifyMap camUtxo
     oldCachedHist <- fromMaybe [] <$> WS.getHistoryCache wid
-    WS.updateHistoryCache wid $ DL.toList camAddedHistory <> oldCachedHist
+    sortedAddedHistory <- sortTxs (DL.toList camAddedHistory)
```

Comment:
Actually, we have idea about transaction order in block - they are sorted _topologically_, which is more substantial than sorting by timestamps (in my opinion). Of course, in cases when transactions are topologically incomparable, they may permute, and in these cases sorting them by timestamp makes sense.

But here you sort all transactions by their timestamps, and these transactions are very likely to not have a timestamp record in metadata yet. You should at least use `_thTimestamp` field of `TxHistoryEntry` in case if there's no `TxMeta` record found - in this case you at least won't accidentally place transactions from old blocks, for which there's no metadata for some reason (it's very possible, afaik: when restoring history from the blockchain in the first time there's no metadata for transactions).

The best solution would require somehow maintaining info about partial order after topsort, but this seems really complicated, so please just consider `_thTimestamp` here and it should be somewhat fine.